src/components/mapgl-layer-icon/mapgl-layer-icon.component.ts

```diff
@@ -125,7 +125,77 @@ export function drawClusterFillIcon(svgNode: SVGElement, colorLegend: Legend) {
       .attr('fill', fourColors[3]).attr('fill-opacity', 0.6)
       .attr('stroke', fourColors[3]).attr('stroke-width', 0.6)
       .attr('y', 3).attr('x', 10);
+}
 
+/**
+ * draws the rectangles icon for feature and feature-metric modes
+ * @param svgNode SVG element on which we append the rectangles using d3.
+ * @param colorLegend Color legend, to give the drawn icons rectagles the same color on the map
+ */
+export function drawFeatureFillIcon(svgNode: SVGElement, colorLegend: Legend) {
+  const colorsList = [];
+  if (colorLegend.type === PROPERTY_SELECTOR_SOURCE.interpolated) {
+    const iv = colorLegend.interpolatedValues;
+    if (iv) {
+      if (iv.length === 1) {
+        for (let i = 0; i < 3; i++) {
+          colorsList.push(iv[0]);
```

Comment:
did you test colorsList.push(iv[0], iv[0], iv[0]); ? (you may also shorten the other conditions if it works)

Reply:
fixed